docs/METRICS.md

```diff
@@ -0,0 +1,108 @@
+
+# Snap Static and Dynamic Metrics
+
+Snap Framework supports two types of metrics. They are static and dynamic metrics. A Snap metric consists of a namespace and value pair.
+
+### Static Metrics
+A namespace having no wildcard and only string literals separated by slashes is a static metric. For example:
+```
+/intel/cassandra/node/zeus/type/Cache/scope/KeyCache/name/Requests/OneMinuteRate
+/intel/cassandra/node/zeus/type/ClientRequest/scope/CASRead/name/Unavailables/FiveMinuteRate
+/intel/cassandra/node/apollo/type/Cache/scope/RowCache/name/Hits/OneMinuteRate
+/intel/cassandra/node/apollo/type/ClientRequest/scope/RangeSlice/name/Latency/OneMinuteRate
+```
+### Dynamic Metrics
+A namespace included at least one wildcard is a dynamic metric. For example:
+```
+/intel/cassandra/node/*/type/*/scope/*/name/*/OneMinuteRate
+/intel/cassandra/node/*/type/*/scope/*/name/*/FiveMinuteRate
+/intel/cassandra/node/*/type/*/keyspace/*/name/*/OneMinuteRate
+/intel/cassandra/node/*/type/*/keyspace/*/name/*/FiveMinuteRate
+```
+### Namespace Element
+Namespace element in Snap is defined as a struct. Both static and dynamic elements share the same definition.
+ ```
+ type NamespaceElement struct {
+	Value       string
+	Description string
+	Name        string
+}
+ ```
+The _`NamespaceElement`_ forms each cell of a namespace and those cells are separated by slashes. 
+
+Create a new instance of a namespace:
+```
+1. // contains all string literals
+2. ns := core.NewNamespace("intel", "cassandra", "node")
+
+3. // contains a wildcard
+4. ns := core.NewNamespace("intel","cassandra", "node", "*")
```

Comment:
(nit) Space is missing after first comma.

But I think would not even advertise this. I would just advertise the `AddDynamicElement` function for the creation of dynamic elements.